src/components/stepforms/DateExtractStepForm.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <div>
+    <step-form-header :cancel="cancelEdition" :title="title" :stepName="this.editedStep.name" />
+    <ColumnPicker
+      id="column"
+      v-model="editedStep.column"
+      name="Column to work on:"
+      :options="columnNames"
+      placeholder="Pick a column"
+      data-path=".column"
+      :errors="errors"
+    />
+    <AutocompleteWidget
+      name="Property to extract"
+      id="operation"
+      :options="operations"
+      :value="currentOperation"
+      @input="updateCurrentOperation"
+      placeholder="Extract operations"
+      :trackBy="`operation`"
+      :label="`label`"
+      data-path=".operation"
+      :errors="errors"
+    />
+    <InputTextWidget
+      id="newColumnName"
+      v-model="editedStep.new_column_name"
+      name="New column name:"
+      :placeholder="newColumnNamePlaceholder"
+      data-path=".new_column_name"
+      :errors="errors"
+    />
+    <step-form-buttonbar :cancel="cancelEdition" :submit="submit" />
+  </div>
+</template>
+
+<script lang="ts">
+import { Prop } from 'vue-property-decorator';
+
+import { StepFormComponent } from '@/components/formlib';
+import ColumnPicker from '@/components/stepforms/ColumnPicker.vue';
+import AutocompleteWidget from '@/components/stepforms/widgets/Autocomplete.vue';
+import InputTextWidget from '@/components/stepforms/widgets/InputText.vue';
+import { generateNewColumnName } from '@/lib/helpers';
+import { DateExtractPropertyStep } from '@/lib/steps';
+
+import BaseStepForm from './StepForm.vue';
+
+type OperationOption = {
+  operation: DateExtractPropertyStep['operation'];
+  label: string;
+};
+
+@StepFormComponent({
+  vqbstep: 'dateextract',
+  name: 'dateextract-step-form',
+  components: {
+    AutocompleteWidget,
+    ColumnPicker,
+    InputTextWidget,
+  },
+})
+export default class DateExtractStepForm extends BaseStepForm<DateExtractPropertyStep> {
+  @Prop({ type: Object, default: () => ({ name: 'dateextract', column: '' }) })
+  initialStepValue!: DateExtractPropertyStep;
+
+  readonly title: string = 'Convert Column From Text to Date';
+
+  readonly operations: OperationOption[] = [
```

Comment:
There are too many operations to display in the UI. We need to keep a direct access to year/month/day/week, and have an "other" that will open the step form on click